### PR TITLE
Skip pairs of zero-byte files.

### PR DESCRIPTION
**Pull request by DanielChicot**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -88,6 +88,12 @@ jobs:
             done
 
             docker-compose up -d dks-standalone-http dks-standalone-https
+
+            while ! docker exec dks-standalone-http cat logs/dks.out | fgrep -q "Started DataKeyServiceApplication"; do
+              echo "Waiting for dks"
+              sleep 2
+            done
```
Comment (dan-hill2802) [inline .circleci/config.yml]: you might want an exit clause like: if time > max_time; exit 1

File: src/main/kotlin/app/batch/S3Reader.kt [hunk 2]
```diff
@@ -97,16 +97,29 @@ class S3Reader(private val s3client: AmazonS3, private val keyPairGenerator: Key
...
-                fileNameFormat.toRegex(),
-                fileNameFormatDataExtension.toRegex(),
-                fileNameFormatMetadataExtension.toRegex())
-
-        val pairs = keyPairs.map {
-            val obj = objectSummaryKeyMap[it.dataKey]
-            val meta = objectSummaryKeyMap[it.metadataKey]
-            S3ObjectSummaryPair(obj, meta)
-        }
+                keyPairGenerator.generateKeyPairs(objectSummaries.map { it.key },
```
Comment (dan-hill2802) [inline src/main/kotlin/app/batch/S3Reader.kt:100]: why has the indentation changed?

Comment (steveburton4) [reply src/main/kotlin/app/batch/S3Reader.kt:100]: Looks right to me now, inline with the others

**Review by steveburton4 — DISMISSED** [review]

File: src/main/kotlin/app/batch/S3Reader.kt (hunk 2 above)
Comment (dan-hill2802) [reply src/main/kotlin/app/batch/S3Reader.kt:100]: how so? The rest of the file is indented less

Comment (dan-hill2802) [reply src/main/kotlin/app/batch/S3Reader.kt:100]: is it because you should indent broken lines?

File: .circleci/config.yml (hunk 1 above)
Comment (DanielChicot) [reply .circleci/config.yml]: I think I should just remove the echo? If a job has no output for 10 minutes circle kills it. Failing that it kills it after 5 hours.

File: src/main/kotlin/app/batch/S3Reader.kt (hunk 2 above)
Comment (DanielChicot) [reply src/main/kotlin/app/batch/S3Reader.kt:100]: this was almost certainly intellij reformatting on commit.

File: .circleci/config.yml (hunk 1 above)
Comment (dan-hill2802) [reply .circleci/config.yml]: If CircleCI has a 10 minute limit then that's a good enough exit clause

**Review by dan-hill2802 — DISMISSED** [review]

**Review by steveburton4 — APPROVED** [review]

**Pull request merged**